### PR TITLE
fix: Auth2 consistent User types

### PR DESCRIPTION
**Pull request by danielgent**

### CODE REVIEW

File: src/adapters/planetscale/users/list.ts [hunk 1]
```diff
@@ -37,37 +36,7 @@ export function listUsers(db: Kysely<Database>) {
...
-        updated_at,
-        logins_count: 0,
-        identities: tags.map((tag) => ({
-          profileData: {
-            email: user.email,
-            user_id: user.id,
-            is_social: true,
-            connection: tag,
-          },
-        })),
```
Comment (danielgent) [inline src/adapters/planetscale/users/list.ts:63]: ahhhh, so I should be doing something here with the `tags`!

We are storing some kind of info I can use for identities :thinking: 

Maybe this should be my next step. It's another missing piece of the jigsaw

File: src/routes/management-api/users.ts [hunk 2]
```diff
@@ -62,11 +65,33 @@ export class UsersMgmtController extends Controller {
...
+        last_login: "",
+        identities: [],
+      };
+    });
+
     if (include_totals) {
-      return result as GetUserResponseWithTotals;
+      // ah but don't we need to actually provide the totals here?
+      // this "as" is going to hide issues
+      const res: GetUserResponseWithTotals = {
```
Comment (danielgent) [inline src/routes/management-api/users.ts:84]: I should copy this on the other routes. no more `as` casting :+1:

File: src/routes/tsoa/login.ts
Comment (danielgent) [inline src/routes/tsoa/login.ts:1]: Pending discussion on the PR this PR is going into, I just commented out everything which makes TypeScript complain :+1:

File: integration-test/management-api/users.spec.ts [hunk 3]
```diff
@@ -44,7 +44,8 @@ describe("users", () => {
     expect(body.length).toBe(0);
   });
 
-  it("should create a new user for a tenant", async () => {
+  // this is different to Auth0 where user_id OR email is required
```
Comment (danielgent) [inline integration-test/management-api/users.spec.ts:47]: A lot of the comments say something similar to this :smile: 

I won't forget to bring this up again anyway

File: src/routes/tsoa/dbconnection.ts [hunk 4]
```diff
@@ -67,7 +67,9 @@ export class DbConnectionController extends Controller {
     if (!user) {
       user = await ctx.env.data.users.create(client.tenant_id, {
         email: body.email,
-        email_verified: false,
+        tenant_id: client.tenant_id,
+        // TBD - this field isn't in SQL
+        // email_verified: false,
```
Comment (danielgent) [inline src/routes/tsoa/dbconnection.ts:72]: Thoughts?

Comment (danielgent) [reply src/routes/tsoa/dbconnection.ts:72]: I'm not actually sure how to check this `dbconnection` route :thinking: 

I'm guessing it's used on the liquidjs login pages

File: src/adapters/planetscale/users/list.ts (hunk 1 above)
Comment (danielgent) [reply src/adapters/planetscale/users/list.ts:63]: I've moved this into the user's management API list route

File: src/routes/management-api/users.ts [hunk 5]
```diff
@@ -62,11 +63,50 @@ export class UsersMgmtController extends Controller {
...
+        last_login: "",
+        identities,
+        // some fields copied from previous adapter/planetscale/list mapping
+        // TODO: store this field in sql
+        email_verified: true,
+        username: user.email,
+        phone_number: "",
+        phone_verified: false,
+        // Deprecated
+        tags,
```
Comment (danielgent) [inline src/routes/management-api/users.ts:94]: as mentioned above, moved from the planetscale list adapter

File: src/adapters/in-memory/users/listUsers.ts [hunk 6]
```diff
@@ -1,25 +1,14 @@
...
-        user_id: user.id,
-        identities: [],
-        logins_count: 0,
-      };
-      return userResponse;
-    });
-
     return {
-      users: usersResponse,
+      users: users,
```
Comment (markusahlstrand) [inline src/adapters/in-memory/users/listUsers.ts:11]: Shorthand :)

File: src/routes/tsoa/dbconnection.ts (hunk 4 above)
Comment (markusahlstrand) [reply src/routes/tsoa/dbconnection.ts:72]: I think we should set it to email_verified false by default.

**Review by markusahlstrand — APPROVED** [review]
Looks good!

File: src/adapters/in-memory/users/listUsers.ts (hunk 6 above)
Comment (danielgent) [reply src/adapters/in-memory/users/listUsers.ts:11]: In the spirit of _everything stylistic should be enforced by a linter_ should I enable this rule? https://eslint.org/docs/latest/rules/object-shorthand

File: src/routes/tsoa/dbconnection.ts (hunk 4 above)
Comment (danielgent) [reply src/routes/tsoa/dbconnection.ts:72]: We need to add this field to the users table then right @markusahlstrand ?

I could do that in another PR

**Pull request merged**